app/controllers/rails_admin/main_controller.rb

```diff
@@ -56,7 +56,7 @@ def respond_to_missing?(sym, include_private)
     end
 
     def back_or_index
-      params[:return_to].presence && params[:return_to].include?(request.host) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
+      params[:return_to].presence && params[:return_to].starts_with?(request.host) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
```

Comment:
But `request.host` is in the form of `www.example.com`. So this won't allow the valid url `https://www.example.com/`, right?

Reply:
You are right, we need to take the scheme into account. I'll fix it. Thx 🙌 !